tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1646,6 +1673,16 @@ void StorageManager::increment_in_progress() {
 }
 
 Status StorageManager::is_array(const URI& uri, bool* is_array) const {
+  // Check if the schema directory exists or not
+  bool is_dir = false;
+  // Since is_dir could return NOT Ok status, we will not use RETURN_NOT_OK here
+  vfs_->is_dir(uri.join_path(constants::array_schema_folder_name), &is_dir);
```

Comment:
catch the return status and modify `if (is_dir) {` to `if (!st.ok() || is_dir) {`

Reply:
changed to if(!st.ok() ||  is_dir). but not very sure if we should return true when status is not ok,